api/sys_mounts.go

```diff
@@ -164,6 +164,7 @@ type MountConfigOutput struct {
 	DefaultLeaseTTL           int      `json:"default_lease_ttl" mapstructure:"default_lease_ttl"`
 	MaxLeaseTTL               int      `json:"max_lease_ttl" mapstructure:"max_lease_ttl"`
 	ForceNoCache              bool     `json:"force_no_cache" mapstructure:"force_no_cache"`
+	DeprecatedPluginName      string   `json:"plugin_name,omitempty" mapstructure:"plugin_name"`
```

Comment:
Personally I would rather have `PluginName` and a comment above documenting that it's deprecated but kept for BC reasons, and to use `Type` instead.